ext/uri/php_lexbor.c

```diff
@@ -0,0 +1,638 @@
+/*
+   +----------------------------------------------------------------------+
+   | Copyright (c) The PHP Group                                          |
+   +----------------------------------------------------------------------+
+   | This source file is subject to version 3.01 of the PHP license,      |
+   | that is bundled with this package in the file LICENSE, and is        |
+   | available through the world-wide-web at the following url:           |
+   | https://www.php.net/license/3_01.txt                                 |
+   | If you did not receive a copy of the PHP license and are unable to   |
+   | obtain it through the world-wide-web, please send a note to          |
+   | license@php.net so we can mail you a copy immediately.               |
+   +----------------------------------------------------------------------+
+   | Authors: Máté Kocsis <kocsismate@php.net>                            |
+   +----------------------------------------------------------------------+
+*/
+
+#include "php.h"
+#include "php_lexbor.h"
+#include "php_uri_common.h"
+#include "Zend/zend_enum.h"
+#include "Zend/zend_smart_str.h"
+#include "Zend/zend_exceptions.h"
+#ifdef HAVE_ARPA_INET_H
+#include <arpa/inet.h>
+#endif
+
+ZEND_TLS lxb_url_parser_t lexbor_parser;
+ZEND_TLS int lexbor_urls;
```

Comment:
```suggestion
ZEND_TLS uint64_t lexbor_urls;
```
I would avoid plain `int` where possible.

Reply:
Since the actual number of URLs doesn't matter at all (only their modulo of `500` is calculated), it should be fine to use a smaller data type like `unsigned short int`.

Reply:
Due to alignment it likely takes up 4 bytes anyway even if you use an unsigned short. It doesn't really matter.